R/input-time-series.R

```diff
@@ -1,131 +1,266 @@
+    dplyr::select(area_id, area_name, area_level, area_level_label,parent_area_id,
+                  area_sort_order, sex, age_group,time_period, year, quarter,
+                  calendar_quarter, dplyr::everything())
+
+  return(art_long)
+}
+
+
+##' Prepare data for ART input time series plots
+##'
+##' Take uploaded ART and shape file paths and format as data which
+##' can be used to draw input time series graphs.
+##'
+##' @param art Path to file containing ART data or ART data object
+##' @param shape Path to file containing geojson areas data or area data object
+##'
+##' @return Data formatted for plotting input time series containing columns
+##' area_id, area_name, area_level, area_level_label, time_period, year,
+##' quarter, plot and value
+##'
+##' @export
+prepare_input_time_series_art <- function(art, shape) {
+
+  ## Check if shape is object or file path
+  if(!inherits(shape, "sf")) {
+    areas <- sf::read_sf(shape) %>% sf::st_drop_geometry()
+  }
+
+  if(inherits(shape, "sf")) {
+    areas <- shape %>% sf::st_drop_geometry()
+  }
+
+  ## Check if art is object or file path
+  if(!inherits(art, c("spec_tbl_df","tbl_df","tbl","data.frame" ))) {
+    art <- readr::read_csv(art, show_col_types = FALSE)
+  }
+
+
+  ## Recursively aggregate ART data up from lowest level of programme data provided
+  # Levels to aggregate up from
+  art_long <- aggregate_art(art, shape)
+  sex_level <- unique(art_long$sex)
+  age_level <- unique(art_long$age_group)
 
   ## Shape data for plot
   art_plot_data <- art_long %>%
-    dplyr::left_join(areas, by = "area_id" ) %>%
-    dplyr::group_by(area_id, area_name, area_level_label, area_level,
-                    parent_area_id, area_sort_order, time_step, time_period) %>%
+    dplyr::group_by(area_id, area_name, area_level, area_level_label,parent_area_id,
+                    area_sort_order,time_period, year, quarter, calendar_quarter) %>%
     dplyr::summarise(
       art_total = sum(art_current, na.rm = TRUE),
+      art_adult = sum(art_current * as.integer(age_group == "Y015_999"), na.rm = TRUE),
       art_adult_f = sum(art_current * as.integer(sex == "female" & age_group == "Y015_999"), na.rm = TRUE),
       art_adult_m = sum(art_current * as.integer(sex == "male" & age_group == "Y015_999"), na.rm = TRUE),
-      art_adult = sum(art_current * as.integer(age_group == "Y015_999"), na.rm = TRUE),
       art_child = sum(art_current * as.integer(age_group == "Y000_014"), na.rm = TRUE),
       .groups = "drop"
     ) %>%
     dplyr::mutate(
       art_adult_sex_ratio = art_adult_f / art_adult_m,
       art_adult_child_ratio = art_adult / art_child,
-      art_prop_u15 = round(art_child/(art_adult+art_child),3)
-    ) %>%
-    tidyr::pivot_longer(cols = dplyr::starts_with("art"),
+      art_prop_u15 = round(art_child/(art_adult+art_child),3),
+    )
+
+  # if art_new column exists in art data, calculate variables
+  if(any(grep("art_new", colnames(art_long)))) {
```

Comment:
@jeffeaton this currently aggregates art_new by area and not over time. Would you like this variable to have a single annual value or should it be iteratively summed up across quarters for each year?